platform/ios/src/MGLMapView.mm

```diff
@@ -941,10 +932,54 @@ - (void)updateConstraintsForOrnament:(UIView *)view
     [constraints addObjectsFromArray:updatedConstraints];
 }
 
+- (BOOL)hasAutomaticallyAdjustContentInset {
+    BOOL automaticallyAdjustContentInset;
+    if (_automaticallyAdjustContentInsetHolder) {
+        automaticallyAdjustContentInset = _automaticallyAdjustContentInsetHolder.boolValue;
+    } else {
+        UIViewController *viewController = [self rootViewController];
+        automaticallyAdjustContentInset = viewController.automaticallyAdjustsScrollViewInsets;
+    }
+    
+    return automaticallyAdjustContentInset;
+}
+
+- (NSLayoutYAxisAnchor *)safeTopAnchor {
+    if ([self hasAutomaticallyAdjustContentInset]) {
+        return self.mgl_safeTopAnchor;
+    } else {
+        return self.topAnchor;
+    }
+}
```

Comment:
Another confusing one for me - is there anyway to combine `safeTapAnchor` and `mgl_safeTopAnchor`?

Reply:
What this method is doing is check if it has automatically adjust content insets. If it has then it uses an anchor that considers safe areas which is `self.mgl_safeTopAnchor` that is in a category, if it is not then uses the current view anchor without considering safe areas.